CMakeLists.txt

```diff
@@ -136,6 +137,18 @@ set(IRODS_COMPILE_DEFINITIONS ${IRODS_COMPILE_DEFINITIONS}
                               BOOST_SYSTEM_NO_DEPRECATED
                               BOOST_CHRONO_DONT_PROVIDES_DEPRECATED_IO_SINCE_V2_0_0
                               BOOST_STACKTRACE_USE_ADDR2LINE
+                              BOOST_ASIO_NO_DEPRECATED
+                              # Manually defined because Boost cannot detect that std::result_of
+                              # has been removed in C++20. Verified that Boost 1.77 and Boost 1.78
+                              # do not handle this yet. Normally, defining BOOST_ASIO_HAS_ macros
+                              # should be avoided except when absolutely required as a workaround
+                              # for the latest version of a compiler or platform.
+                              #
+                              # See the following for more information:
+                              #
+                              #   https://www.boost.org/doc/libs/1_77_0/doc/html/boost_asio/using.html
+                              #
+                              BOOST_ASIO_HAS_STD_INVOKE_RESULT
```

Comment:
this should be captured in an issue.

once boost is able to detect `std::invoke_result`, this macros should be removed.